rdr_service/dao/study_nph_dao.py

```diff
@@ -73,26 +73,46 @@ def from_client_json(self):
         pass
 
     def get_consents_subquery(self):
+        """
+        Returns a subquery that gets the latest consents for each participant grouped by event_type_id.
+        """
+        ce2 = aliased(ConsentEvent)
         with self.session() as session:
-            return session.query(
-                Participant.id.label('consent_pid'),
-                func.json_object(
-                    'consent_json',
-                    func.json_arrayagg(
-                        func.json_object(
-                            "value", ConsentEventType.source_name,
-                            "time", ConsentEvent.event_authored_time,
-                            "opt_in", ConsentEvent.opt_in,
-                        )
-                    ), type_=JSON
-                ).label('consent_status'),
-            ).join(
-                ConsentEvent,
-                ConsentEvent.participant_id == Participant.id
-            ).join(
-                ConsentEventType,
-                ConsentEventType.id == ConsentEvent.event_type_id,
-            ).group_by(Participant.id).subquery()
+            return (
+                session.query(
+                    Participant.id.label("consent_pid"),
+                    func.json_object(
+                        "consent_json",
+                        func.json_arrayagg(
+                            func.json_object(
+                                "value",
+                                ConsentEventType.source_name,
+                                "time",
+                                ConsentEvent.event_authored_time,
+                                "opt_in",
+                                ConsentEvent.opt_in,
+                            )
+                        ),
+                        type_=JSON,
+                    ).label("consent_status"),
+                )
+                .join(ConsentEvent, ConsentEvent.participant_id == Participant.id)
+                .join(
+                    ConsentEventType,
+                    ConsentEventType.id == ConsentEvent.event_type_id,
+                )
+                .outerjoin(
+                    ce2,
+                    and_(
+                        ce2.participant_id == ConsentEvent.participant_id,
+                        ce2.event_type_id == ConsentEvent.event_type_id,
+                        ce2.id > ConsentEvent.id,
+                    ),
+                )
```

Comment:
:nit For better readability that will mimic the rest of these types of outerjoins in this dao, can we move the alias to the left side of the assignment?
ex: `ConsentEvent.participant_id == ce2.participant_id`